.github/workflows/test-integration-vizro-ai.yml

```diff
@@ -47,55 +70,76 @@ jobs:
+        # https://stackoverflow.com/questions/66025220/paired-values-in-github-actions-matrix
+        # https://github.com/orgs/community/discussions/26253
+        # https://github.com/actions/runner/issues/1985
+        config:
           - python-version: "3.9"
             hatch-env: all.py3.9
+            os: ubuntu-latest
           - python-version: "3.10"
             hatch-env: all.py3.10
+            os: ubuntu-latest
           - python-version: "3.11"
             hatch-env: all.py3.11
+            os: ubuntu-latest
           - python-version: "3.12"
             hatch-env: all.py3.12
+            os: ubuntu-latest
           - python-version: "3.9"
             hatch-env: lower-bounds
+            os: ubuntu-latest
             label: lower bounds
+          - python-version: "3.12"
+            hatch-env: all.py3.12
+            os: windows-latest
+            label: Windows
+        is_pr:
+          - ${{ github.event_name != 'pull_request' }}
+        exclude:
+          - is_pr: true
+            config:
+              python-version: "3.12"
+              hatch-env: all.py3.12
+              os: windows-latest
+              label: Windows
 
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Python ${{ matrix.python-version }}
+      - name: Set up Python ${{ matrix.config.python-version }}
         uses: actions/setup-python@v5
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: ${{ matrix.config.python-version }}
 
       - name: Install Hatch
         run: pip install hatch
 
       - name: Show dependency tree
-        run: hatch run ${{ matrix.hatch-env }}:pip tree
+        run: hatch run ${{ matrix.config.hatch-env }}:pip tree
 
-      - name: Run vizro-ai integration tests with pypi vizro
-        run: |
-          export OPENAI_API_KEY=${{ secrets.OPENAI_API_KEY }}
-          export OPENAI_API_BASE=${{ secrets.OPENAI_API_BASE }}
-          export VIZRO_TYPE=pypi
-          hatch run ${{ matrix.hatch-env }}:test-integration
+      - name: Run vizro-ai integration tests with PyPI vizro
+        run: hatch run ${{ matrix.config.hatch-env }}:test-integration
+        env:
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+          OPENAI_API_BASE: ${{ secrets.OPENAI_API_BASE }}
+          VIZRO_TYPE: pypi
 
       - name: Run vizro-ai integration tests with local vizro
         run: |
-          export OPENAI_API_KEY=${{ secrets.OPENAI_API_KEY }}
-          export OPENAI_API_BASE=${{ secrets.OPENAI_API_BASE }}
-          export VIZRO_TYPE=local
-          cd ../vizro-core
-          hatch build
-          cd ../vizro-ai
-          hatch run ${{ matrix.hatch-env }}:pip install ../vizro-core/dist/vizro*.tar.gz
-          hatch run ${{ matrix.hatch-env }}:test-integration
+          hatch run ${{ matrix.config.hatch-env }}:pip install ../vizro-core
+          hatch run ${{ matrix.config.hatch-env }}:test-integration
```

Comment:
FYI @l0uden I tidied this up by just installing `vizro-core` directly - no need to build it first.